### DISCUSSION TITLE
my freeCodeCamp tutorial on installing Python using Rye

### DISCUSSION THREAD

**Discussion by DanielKehoe** [astral-sh/rye#1067] [Show and tell]
I've written a tutorial for freeCodeCamp titled [How to Install Python on a Mac](https://www.freecodecamp.org/news/how-to-install-python-on-a-mac/) that champions the use of Rye. 

The freeCodeCamp tutorial is based on a series or articles I wrote for mac.install.guide on [Mac Python](https://mac.install.guide/python/) including [Install Python with Rye](https://mac.install.guide/python/install) and [Use Rye for Python](https://mac.install.guide/python/use-rye).

I welcome any feedback or corrections as the articles are starting to get a bit of traffic. I'm daniel@danielkehoe.com.

**Comment by paolodina** [comment]
I haven’t read the other articles, but the freeCodeCamp article dated May 9th should not reference an outdated version of rye.

```bash
rye --version
rye 0.26.0 # <-- old
```

**Comment by DanielKehoe** [comment]
I'm sad to see the URL I used for Rye in the freeCodeCamp article is no longer working. In the article I linked to https://rye-up.com/ but now I'm seeing "rye-up.com refused to connect." So far, to date 2696 people have viewed the freeCodeCamp article. But I think they won't be able to try Rye unless they do a Google search. And a broken website may discourage some interested users. Unfortunately, it's not possible to change freeCodeCamp articles after they have been published. I wonder if someone can set up a redirect on https://rye-up.com/ to the working https://rye.astral.sh/ website?

**Reply by trag1c** [reply]
From #1113:
> We're having issues with `rye-up.com`. We're trying to recover it but need to work with the registrar. In the interim, I'm going to move the docs and installers to `rye.astral.sh`. When `rye-up.com` is recovered, we can redirect, either from `rye.astral.sh` to `rye-up.com` or vice versa.

Also honestly I feel like that's something that should be brought up with freeCodeCamp to allow edits like this (even if via their support team or something) rather than each individual project.

EDIT:
And it seems like that is (or at least was) doable. From [their forum](https://forum.freecodecamp.org/t/editing-an-article/426559):
> You can submit your drafts by emailing a link to them to [editorial@freecodecamp.org](mailto:editorial@freecodecamp.org). And if you need any edits to your articles that are already live, just email us there and we’ll make the edits for you real quick.

**Reply by DanielKehoe** [reply]
I just checked and saw you now have a redirect from `rye-up.com` to `rye.astral.sh`. Great!

Also, I got the editors at freeCodeCamp to update the URL in the article. As an infrequent contributor to freeCodeCamp, I didn't know there was a procedure available for article edits. So all good now! And may Rye continue to grow! I know I'm not the only one who appreciates the developers' efforts to improve the Python DX.

**Reply by charliermarsh** [reply]
Thanks @DanielKehoe! Sorry for the disruption here.

**Comment by rishabhtpt** [comment]
Python is used in so many areas – from building websites and analyzing data to creating AI and automating repetitive tasks. A good [Python Tutorial | Python Programming Language](https://www.tpointtech.com/python-tutorial) will give you the foundation to explore these exciting fields.
